stdlib/entity/resource.lua

```diff
@@ -0,0 +1,182 @@
+    fail_if_missing(surface, "missing surface")
+    fail_if_missing(position, "missing position")
+
+    -- get the initial resource tile if there is one at the given position
+    local all_resource_entities = Resource.get_resources_at(surface, position)
+    local all_resource_types = Resource.get_resource_types(all_resource_entities)
+    local resource_patches = {}
+
+    for _,type in pairs(all_resource_types) do
+        local resource_patch = Resource.get_resource_patch_at(surface, position, type)
+        resource_patches[type] = resource_patch
+    end
+
+    return resource_patches
+end
+
+--- From the resources at the given surface and position, return all connected (horizontally, vertically and diagonally) resource entities of the specified type.
+-- (For now, just ore patches. Problems arise when a single resource entity spans multiple tiles).
+-- (Note that the implementation is not stable: If a resource entity reference changes during the search, the old and the new version of the entity might be included)
+-- @tparam LuaSurface surface the surface of the position
+-- @tparam LuaPosition position the position to check
+-- @tparam string type the resource type, i.e. "iron-ore"
+-- @treturn {nil|LuaEntity,...} an array containing all resources in the resource patch, or an empty array if there are no resources there
+function Resource.get_resource_patch_at(surface, position, type)
+    fail_if_missing(surface, "missing surface")
+    fail_if_missing(position, "missing position")
+    fail_if_missing(position, "missing ore name")
+
+    -- get the initial resource tile if there is one at the given position
+    local all_resource_entities = Resource.get_resources_at(surface, position)
+    local filtered_resource_entities = Resource.filter_resources(all_resource_entities, {type})
+
+    if #filtered_resource_entities == 0 then
+        return {}
+    end
+
+    -- for the search, keep track of the relevant entities and tiles visited
+    -- we use the entities as keys to prevent having a single entity multiple times in the list
+    local resource_patch = {}
+    local visited_tiles = {}
+
+    local initial_tile = Tile.from_position(filtered_resource_entities[1].position)
+
+    -- do a BFS starting from the initial tile
+    local search_queue = Queue.new()
+    Queue.push_right(search_queue, initial_tile)
+
+    while not Queue.is_empty(search_queue) do
+        local current_tile = Queue.pop_left(search_queue)
+        local current_entities = Resource.get_resources_at(surface, current_tile)
+        local current_tile_index = Tile.get_index(current_tile)
+        visited_tiles[current_tile_index] = true
+
+        local filtered_current_entities = Resource.filter_resources(current_entities, {type})
+
+        if #filtered_current_entities ~= 0 then
+            -- this tile belongs to the ore patch, add the resources
+            table.merge(resource_patch, table.invert(filtered_current_entities))
```

Comment:
I implemented it. :)
https://github.com/Afforess/Factorio-Stdlib/pull/93/commits/65d7c27d872cc66b43b50c13e1f853369664c580#diff-fa80bf2caece2b96c15b11315ccabce7